src/core/mod.rs

```diff
@@ -920,6 +927,11 @@ impl JwsSigningAlgorithm<CoreJsonWebKeyType> for CoreJwsSigningAlgorithm {
                 hasher.update(bytes);
                 hasher.finalize().to_vec()
             }
+            CoreJwsSigningAlgorithm::EdDsaEd25519 => {
```

Comment:
I believe it should be using SHA-512: https://www.rfc-editor.org/rfc/rfc8032#section-5.1. since that's the hash function used by the signature scheme, we should be using the same one here for computing `at_hash` and `c_hash`.

Reply:
hello @ramosbugs, I think the message is being hashed with `sha2-512` by default before signing with the key, I think it is the "prehash" from [section-5.1](https://www.rfc-editor.org/rfc/rfc8032#section-5.1) is implemented according to [this from ed25519_dalek](https://github.com/dalek-cryptography/ed25519-dalek/blob/main/src/signing.rs#L508)

Reply:
right, this code isn't used for signing JWTs. it's used for computing the [`at_hash`](https://openid.net/specs/openid-connect-core-1_0.html#CodeIDToken) and `c_hash` values, which is independent of the JWT signing and verification process.

Reply:
it just happens that the hash function used for computing those claims is supposed to be the same one that the signing algorithm uses, and in this case I believe that's SHA-512. otherwise, we wouldn't be able to compute these claims when using EdDSA, which doesn't seem right.

Reply:
oh is it we need to hash it again?

Reply:
Hopefully this helps clarify:
https://github.com/ramosbugs/openidconnect-rs/blob/67474002298c5b818adfcd9f74c898adf42758c6/src/id_token.rs#L87-L98